metadata.json

```diff
@@ -19,7 +20,7 @@
   "requirements": [
     {
       "name": "puppet",
-      "version_requirement": ">= 4.7.0 < 5.0.0"
+      "version_requirement": ">= 4.7.0 < 6.0.0"
```

Comment:
I'm not going to do this right now. I will make a new PR with more improvements which are unrelated to the original PR.

There are a lot of things to fix and I'm don't even think that the class structure of the module is any good.